### PR TITLE
Create field

### PR DESCRIPTION
**Pull request by Hairash**
#5
Create hex grid and draw it on the page.

https://github.com/Miol-Mor/battle-game/wiki/Frontend#api

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,10 @@ backend/Cargo.lock
 
 # These are backup files generated by rustfmt
 backend/**/*.rs.bk
+
+# Files, added by Idea and VSCode IDEs
+.idea
+.vscode
+
+# Files, added by MacOS
+.DS_Store
```
Comment (bronvic) [inline .gitignore:17]: As i remember, `*.min.js` and `*.min.js.map` are autogenerated files and they should be ignored

File: frontend/script.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+let type = "WebGL"
+if(!PIXI.utils.isWebGLSupported()){
+    type = "canvas"
+}
+PIXI.utils.sayHello(type);
+
+window.onload = start;
+
+function start() {
+    //Create a Pixi Application
```
Comment (bronvic) [inline frontend/script.js:10]: I think it's obvious, we can remove this comment

File: frontend/script.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    let app = new PIXI.Application({ 
+        width: 256,         // default: 800
+        height: 256,        // default: 600
+        antialias: true,    // default: false
+        transparent: false, // default: false
+        resolution: 1       // default: 1
+    });
+    app.renderer.backgroundColor = 0xd6b609;
+
+    app.renderer.autoDensity = true; // autoResize is deprecated
```
Comment (bronvic) [inline frontend/script.js:20]: What is comment about autoResize here?

File: frontend/script.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    type = "canvas"
+}
+PIXI.utils.sayHello(type);
+
+window.onload = start;
+
+function start() {
+    //Create a Pixi Application
+    let app = new PIXI.Application({ 
+        width: 256,         // default: 800
```
Comment (bronvic) [inline frontend/script.js:12]: Maybe remove information about default values?

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,10 @@ backend/Cargo.lock
 
 # These are backup files generated by rustfmt
 backend/**/*.rs.bk
+
+# Files, added by Idea and VSCode IDEs
```
[the remark this answers is not in the document]
Comment (Hairash) [reply .gitignore:12]: Ok

File: frontend/grid.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
+class Hex_grid {
+    BORDER_COLOR = 0x000000;
+    BORDER_WIDTH = 1;
+    FILL_COLOR = 0x1020b0;
+
+    // stage - where to draw grid (PIXI.Container)
+    // m, n - number of rows and columns in grid
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:7]: ok

File: frontend/grid.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    // cellsize - size of hexcell
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:8]: ok

File: frontend/grid.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    // x_offset, y_offset - left and top offsets inside stage
+    // cells - matrix of cells (call - cells[y][x])
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:10]: ok

File: frontend/grid.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        this.n = n;
+        this.cell_size = cell_size;
+        this.x_offset = x_offset;
+        this.y_offset = y_offset;
+        this.cells = [];
+    }
+    
+
+    // returns hex object with side = r
+    create_hex(r) {
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:23]: ok, but I left comment

File: frontend/grid.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        let hex = new PIXI.Graphics();
+        
+        hex.lineStyle(this.BORDER_WIDTH, this.BORDER_COLOR);
+
+        //Use `drawPolygon` to define the hex as a path array of x/y positions
+        let points = [
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:29]: ok

File: .gitignore (hunk 1 above)
Comment (Hairash) [reply .gitignore:17]: No.
pixi.min.js is a pixi.js but without whitespace characters.
So pixi.js weights 14Mb, while pixi.min.js - 3Mb.
pixi.min.js.map - I don't know, what is it. You can try delete it.

File: frontend/script.js (hunk 2 above)
Comment (Hairash) [reply frontend/script.js:10]: ok

File: frontend/script.js (hunk 3 above)
Comment (Hairash) [reply frontend/script.js:20]: There was ```autoResize method```, but I changed it and there is a reason.
I will remove this comment.

File: frontend/script.js (hunk 4 above)
Comment (Hairash) [reply frontend/script.js:12]: ok

File: frontend/grid.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    // draws grid on the stage
+    draw() {
+        let grid_container = new PIXI.Container();
+        this.stage.addChild(grid_container);
+        grid_container.position.set(this.x_offset, this.y_offset);
+
+        let r = this.cell_size;
+        let n = this.n,
+            m = this.m;
+        let cells = this.cells;
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:55]: Ok, I can remove variables ```row_n``` and ```col_n```, but I really don't want to type ```this.cell_size``` a lot of times.

File: frontend/grid.js [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+            - r * Math.sqrt(3) / 2, r / 2,
+            0, r,
+            r * Math.sqrt(3) / 2, r / 2,
+            r * Math.sqrt(3) / 2, - r / 2,
+            0, - r,
+            - r * Math.sqrt(3) / 2, - r / 2
+        ];
+        hex.drawPolygon(points);
+
+        let hex_obj = {};
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:39]: Ok.
Create class ```Hex``` extends ```PIXI.Graphics()```

File: frontend/grid.js [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+        hex_obj.hex = hex; // Graphics object
+        hex_obj.points = points; // array of points
```
[the remark this answers is not in the document]
Comment (Hairash) [reply frontend/grid.js:41]: ok

**Pull request merged**